beacon-chain/sync/fork_watcher.go

```diff
@@ -12,20 +12,23 @@ import (
 // Is a background routine that observes for new incoming forks. Depending on the epoch
 // it will be in charge of subscribing/unsubscribing the relevant topics at the fork boundaries.
 func (s *Service) forkWatcher() {
-	<-s.initialSyncComplete
+	// At startup, launch registration and peer discovery loops, and register rpc stream handlers.
+	startEntry := params.GetNetworkScheduleEntry(s.cfg.clock.CurrentEpoch())
+	s.registerSubscribers(startEntry)
+
 	slotTicker := slots.NewSlotTicker(s.cfg.clock.GenesisTime(), params.BeaconConfig().SecondsPerSlot)
 	for {
 		select {
 		// In the event of a node restart, we will still end up subscribing to the correct
 		// topics during/after the fork epoch. This routine is to ensure correct
 		// subscriptions for nodes running before a fork epoch.
-		case currSlot := <-slotTicker.C():
-			currEpoch := slots.ToEpoch(currSlot)
-			if err := s.registerForUpcomingFork(currEpoch); err != nil {
+		case <-slotTicker.C():
+			current := s.cfg.clock.CurrentEpoch()
```

Comment:
Why not using the slot given by `<-slotTicker.C()`? It seems more expensive not to use it.

Reply:
`slotTicker` chooses to guarantee that we get a message for every slot at the cost of potentially sending old slots over the channel. As a defensive programming measure I never use the slot value from slot ticker in this kind of situation as I think it's better to always ensure you are acting on the real current slot/epoch. That to me is worth the trivial expense of computing the current slot from the time.Now.